pkg/clusteragent/custommetrics/provider.go

```diff
@@ -0,0 +1,134 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubeapiserver
+
+package custommetrics
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/provider"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/metrics/pkg/apis/custom_metrics"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/hpa"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type externalMetric struct {
+	info  provider.ExternalMetricInfo
+	value external_metrics.ExternalMetricValue
+}
+
+type datadogProvider struct {
+	client dynamic.ClientPool
+	mapper apimeta.RESTMapper
+
+	values          map[provider.CustomMetricInfo]int64
+	externalMetrics []externalMetric
+	resVersion      string
+	hpaClient       *hpa.HPAWatcherClient
+}
+
+// NewDatadogProvider creates a Custom Metrics and External Metrics Provider.
+func NewDatadogProvider(client dynamic.ClientPool, mapper apimeta.RESTMapper, hpaCl *hpa.HPAWatcherClient) provider.MetricsProvider {
+	return &datadogProvider{
+		client:    client,
+		mapper:    mapper,
+		values:    make(map[provider.CustomMetricInfo]int64),
+		hpaClient: hpaCl,
+	}
+}
+
+// GetRootScopedMetricByName - Not implemented
+func (p *datadogProvider) GetRootScopedMetricByName(groupResource schema.GroupResource, name string, metricName string) (*custom_metrics.MetricValue, error) {
+	return nil, fmt.Errorf("Not Implemented - GetRootScopedMetricByName")
+}
+
+// GetRootScopedMetricBySelector - Not implemented
+func (p *datadogProvider) GetRootScopedMetricBySelector(groupResource schema.GroupResource, selector labels.Selector, metricName string) (*custom_metrics.MetricValueList, error) {
+	return nil, fmt.Errorf("Not Implemented - GetRootScopedMetricBySelector")
+}
+
+// GetNamespacedMetricByName - Not implemented
+func (p *datadogProvider) GetNamespacedMetricByName(groupResource schema.GroupResource, namespace string, name string, metricName string) (*custom_metrics.MetricValue, error) {
+	return nil, fmt.Errorf("Not Implemented - GetNamespacedMetricByName")
+}
+
+// GetNamespacedMetricBySelector - Not implemented
+func (p *datadogProvider) GetNamespacedMetricBySelector(groupResource schema.GroupResource, namespace string, selector labels.Selector, metricName string) (*custom_metrics.MetricValueList, error) {
+	return nil, fmt.Errorf("Not Implemented - GetNamespacedMetricBySelector")
+}
+
+// ListAllMetrics reads from a ConfigMap, similarly to ListExternalMetrics
+func (p *datadogProvider) ListAllMetrics() []provider.CustomMetricInfo {
+	return nil
```

Comment:
`return nil, fmt.Errorf("Not Implemented - ListAllMetrics")` ?

Reply:
Yes, I'd rather not return an error as it gets logged every 20-30 seconds in the log of the DCA. I'll add the TODO